File: IP/3rd/sampleprogram/4-8-3.cpp
#define _USE_MATH_DEFINES
#include <iostream>
#include <cmath>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;

int main()
{
  string file_src = "src.png";
  Mat img_gray = imread(file_src, 0);
  Mat img_src = imread(file_src, 1);
  namedWindow("src", WINDOW_AUTOSIZE);
  namedWindow("gray", WINDOW_AUTOSIZE);

  int x = 10, y = 50, v = 255, r = 255, g = 255, b = 0;

  // グレースケール画像の場合
  cout << (int)img_gray.at<uchar>(y, x) << endl;
  img_gray.at<uchar>(y, x) = 255;
  cout << (int)img_gray.at<uchar>(y, x) << endl;

  // カラー画像の場合
  cout << img_src.at<Vec3b>(y, x) << endl;
  img_src.at<Vec3b>(y, x) = Vec3b(b, g, r);
  cout << img_src.at<Vec3b>(y, x) << endl;

  imshow("src", img_src);
  imshow("gray", img_gray);
  waitKey(0);
  return 0;
}

File: OpenCVによる画像処理入門/第3版/第1刷/sampleprogram/4-8-3.py
import cv2

file_src = 'src.png'
img_gray = cv2.imread(file_src, 0)
img_src = cv2.imread(file_src, 1)
cv2.namedWindow('gray')
cv2.namedWindow('src')

x = 10
y = 50
v = 255
r = 255
g = 255
b = 0

# グレースケール画像の場合
print(img_gray[y, x])
img_gray[y, x] = v
print(img_gray[y, x])

# カラー画像の場合
print(img_src[y, x])
img_src[y, x] = [b, g, r]
print(img_src[y, x])

cv2.imshow('gray', img_gray)
cv2.imshow('src', img_src)
cv2.waitKey(0)
cv2.destroyAllWindows()
